src/ape/managers/networks.py

```diff
@@ -90,17 +95,24 @@ def fork(
               When ``None``, returns the default provider.
             provider_settings (dict, optional): Settings to apply to the provider. Defaults to
               ``None``.
+            block_number (Optional[int]): Optionally specify the block number you wish to fork.
+              Negative block numbers are relative to HEAD. Defaults to the configured fork
+              block number or HEAD.
 
         Returns:
             :class:`~ape.api.networks.ProviderContextManager`
         """
-        forked_network = self.ecosystem.get_network(f"{self.network.name}-fork")
+        try:
+            forked_network = self.ecosystem.get_network(f"{self.network.name}-fork")
+        except NetworkNotFoundError as err:
+            raise NetworkError(f"Unable to fork network '{self.network.name}'.") from err
+
         provider_settings = provider_settings or {}
 
         if block_number is not None:
             # Negative block_number means relative to HEAD
             if block_number < 0:
-                block_number = self.provider.get_block("latest").number + block_number
+                block_number = max(0, self.provider.get_block("latest").number)
```

Comment:
bugfix here: when accidentally forking past genesis, it would still be negative.. 
now you are just bound to genesis

hmm, wondering if we should do a warning here?

Reply:
ok exception better, did that instead